Compilation problems with clang-16 compiler

Trying to build audit-userspace project version on my ubuntu-24 with clang-16, I got a problem, causes by declaration of variable in switch statement.

Label case `FA_EMAIL` is attached to the declaration of variable `content` and `subject`. In C language declarations are not statements. They cannot be labeled. And this is what causes the error when this code is interpreted as C code.

Possible fix - add extra `{}` in this case statement to limit the scope of these variables:

```diff
diff --git a/src/auditd-event.c b/src/auditd-event.c
index a6eeb2c1..e20b8ea5 100644
--- a/src/auditd-event.c
+++ b/src/auditd-event.c
@@ -875,6 +875,7 @@ static void do_space_left_action(int admin)
                        }
                        break;
                case FA_EMAIL:
+               {
                        char content[512];
                        const char *subject;
 
@@ -894,6 +895,7 @@ static void do_space_left_action(int admin)
                        sendmail(subject, content, config->action_mail_acct);
                        audit_msg(LOG_ALERT, "%s", content);
                        break;
+               }
                case FA_EXEC:
                        // Close the logging file in case the script zips or
                        // moves the file. We'll reopen in sigusr2 handler
```

Also, add extra `;` fixes the error:

```diff
diff --git a/src/auditd-event.c b/src/auditd-event.c
index a6eeb2c1..06814526 100644
--- a/src/auditd-event.c
+++ b/src/auditd-event.c
@@ -874,7 +874,7 @@ static void do_space_left_action(int admin)
                                rotate_logs(0, 0);
                        }
                        break;
-               case FA_EMAIL:
+               case FA_EMAIL:;
                        char content[512];
                        const char *subject;
```

Steps to reproduce:

1. Configure and make project this way:
```shell
export CC=clang-16
export CXX=clang++-16
export CFLAGS="-g -O0"
export CPP="$CC -E"
export CPPFLAGS="$CFLAGS"

autoreconf -f --install

CC=$CC CPP=$CPP CFLAGS=$CFLAGS CPPFLAGS=$CPPFLAGS ./configure --with-python3=yes --with-apparmor --with-libwrap --enable-gssapi-krb5 --with-libcap-ng --with-arm --with-aarch64 --host=x86_64 --without-golang --with-io_uring --enable-static=yes --enable-shared=no --with-pic=yes
make
```

2. Get an error:
```shell
1 warning generated.
mv -f .deps/auditd-auditd.Tpo .deps/auditd-auditd.Po
clang-16 -DHAVE_CONFIG_H -I. -I..  -I.. -I../lib -I../src/libev -I../auparse -I../audisp -I../common -g -O0 -fPIE -DPIE -g -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pthread -Wno-pointer-sign -Wno-format-truncation -Wno-unused-but-set-variable -g -O0 -MT auditd-auditd-event.o -MD -MP -MF .deps/auditd-auditd-event.Tpo -c -o auditd-auditd-event.o `test -f 'auditd-event.c' || echo './'`auditd-event.c
warning: unknown warning option '-Wno-format-truncation' [-Wunknown-warning-option]
auditd-event.c:878:4: error: expected expression
                        char content[512];
                        ^
auditd-event.c:883:30: error: use of undeclared identifier 'content'; did you mean 'connect'?
                                snprintf(content, sizeof(content),
                                                         ^~~~~~~
                                                         connect
```

--- a/src/auditd-event.c
+++ b/src/auditd-event.c
@@ -875,6 +875,7 @@ static void do_space_left_action(int admin)
 			}
 			break;
 		case FA_EMAIL:
+		{
 			char content[512];
 			const char *subject;
 
@@ -894,6 +895,7 @@ static void do_space_left_action(int admin)
 			sendmail(subject, content, config->action_mail_acct);
 			audit_msg(LOG_ALERT, "%s", content);
 			break;
+		}
 		case FA_EXEC:
 			// Close the logging file in case the script zips or
 			// moves the file. We'll reopen in sigusr2 handler

